### ISSUE TITLE
Is passing a ReadableStream to oboe under browser environment (will be) possible?

### ISSUE THREAD

**Issue by chicoxyzzy** [jimhigson/oboe.js#55]
As I know browsers' WebSocket APIs are message-based so we can only receive messages not frames (and not streams respectively cause there is no actual chunks of data). Does it mean that there is no profit using oboe with WebSockets? Will [passing a ReadableStream](http://oboejs.com/api#byo-stream) to oboe under browser environment be possible after browsers will support [Streams](https://streams.spec.whatwg.org/#rs) and [Streams API](https://streams.spec.whatwg.org/#rs)?

**Comment by jimhigson** [comment]
That sounds perfectly doable. 

The XHR is wrapped to work (sort of) like a stream in [streamingHttp.browser.js](https://github.com/jimhigson/oboe.js/blob/master/src/streamingHttp.browser.js). It wouldn't be a lot of work to bring a little of the (https://github.com/jimhigson/oboe.js/blob/master/src/streamingHttp.node.js)[node equivalent] into the browser.

Is this something that would be useful right now?

**Comment by chicoxyzzy** [comment]
Yes it will be definitely usefull

**Comment by chicoxyzzy** [comment]
But wait. Browsers can't receive WebSocket messages progressively now.

**Comment by chicoxyzzy** [comment]
@jimhigson am I right? And what are the benefits of using oboe in that case? I believe that till we have whole message parsing it once is better than parsing it progressively with oboe.

**Comment by bitinn** [comment]
I would like to see oboe support this on client-side as well, with Fetch API and Stream API are coming together, it would be nice for Oboe to accept a body stream from Fetch response, see this sample repo:

https://github.com/domenic/streams-demo/blob/gh-pages/index.html#L101

`getReader()` will create a readable byte stream:

https://github.com/whatwg/streams/blob/master/docs/ReadableByteStream.md#getreader

And Oboe will be decoding and wrapping the result using its existing api, you may even modularize oboe further so xhr is optional.

Though it may eventually make Oboe look more like a glue and syntax sugar on top of clarinet parser/native stream/fetch api, that's still a noble goal, right? :)

**Comment by JuanCaicedo** [comment]
@bitinn Is that similar to #74?

**Comment by bitinn** [comment]
@JuanCaicedo yep, more or less the same feature request. Though instead of supporting fetch syntax, I am advocating that we modularize oboe.js, to allow fetch api to handle the request separately, and only pass the body stream to oboe.js (much like how you would use it [in node.js environment](http://oboejs.com/examples), see read from node.js stream section)
